app/models/health/soap/mass_health.rb

```diff
@@ -0,0 +1,217 @@
+        else
+          return SoapResponse,new(self, [ message.decoded, nil ])
+        end
+      end
+    end
+
+    private def generic_results_retrieval_request_xml(payload_type, payload_id)
+      timestamp = Time.now.utc.iso8601
+      <<~HEREDOC
+        <cor:COREEnvelopeBatchResultsRetrievalRequest>
+           <PayloadType>#{payload_type}</PayloadType>
+           <ProcessingMode>Batch</ProcessingMode>
+           <PayloadID>#{payload_id}</PayloadID>
+           <TimeStamp>#{timestamp}</TimeStamp>
+           <SenderID>#{@config.sender}</SenderID>
+           <ReceiverID>#{@config.receiver}</ReceiverID>
+           <CORERuleVersion>2.2.0</CORERuleVersion>
+        </cor:COREEnvelopeBatchResultsRetrievalRequest>
+      HEREDOC
+    end
+
+    private def realtime_eligibility_inquiry_request_xml(edi_doc)
+      timestamp = Time.now.utc.iso8601
+      payload_id = SecureRandom.uuid
+      <<~HEREDOC
+        <cor:COREEnvelopeRealTimeRequest>
+           <PayloadType>X12_270_Request_005010X279A1</PayloadType>
+           <ProcessingMode>RealTime</ProcessingMode>
+           <PayloadID>#{payload_id}</PayloadID>
+           <TimeStamp>#{timestamp}</TimeStamp>
+           <SenderID>#{@config.sender}</SenderID>
+           <ReceiverID>#{@config.receiver}</ReceiverID>
+           <CORERuleVersion>2.2.0</CORERuleVersion>
+           <Payload><![CDATA[#{edi_doc}]]></Payload>
+        </cor:COREEnvelopeRealTimeRequest>
+      HEREDOC
+    end
+
+    private def batch_eligibility_request_xml(attachment)
+      timestamp = Time.now.utc.iso8601
+      payload_id = SecureRandom.uuid
+      checksum = Digest::SHA1.hexdigest(attachment)
+      <<~HEREDOC
+        <cor:COREEnvelopeBatchSubmission>
+          <PayloadType>X12_270_Request_005010X279A1</PayloadType>
+          <ProcessingMode>Batch</ProcessingMode>
+          <PayloadID>#{payload_id}</PayloadID>
+          <PayloadLength>#{attachment.length}</PayloadLength>
+          <TimeStamp>#{timestamp}</TimeStamp>
+          <SenderID>#{@config.sender}</SenderID>
+          <ReceiverID>#{@config.receiver}</ReceiverID>
+          <CORERuleVersion>2.2.0</CORERuleVersion>
+          <CheckSum>#{checksum}</CheckSum>
+           <Payload>
+            <xop:Include href="cid:attachment" xmlns:xop="http://www.w3.org/2004/08/xop/include"/>
+           </Payload>
+        </cor:COREEnvelopeBatchSubmission>
+      HEREDOC
+    end
+
+    private def request_xml(body)
+      <<~HEREDOC
+        <soap:Envelope xmlns:cor="http://www.caqh.org/SOAP/WSDL/CORERule2.2.0.xsd" xmlns:soap="http://www.w3.org/2003/05/soap-envelope">
+          <soap:Header>
+            #{wsse_header_xml}
+          </soap:Header>
+          <soap:Body>
+            #{body}
+          </soap:Body>
+        </soap:Envelope>
+      HEREDOC
+    end
+
+    private def request_with_attachment_xml(body, attachment, boundary)
+      # MIME is very picky about whitespace, so, it is important that this not be indented
+      <<~HEREDOC
```

Comment:
This is totally fine, but I believe the ~ HEREDOC eats whitespace.
https://www.rubyguides.com/2018/11/ruby-heredoc/